src/libpmemobj/memblock.c

```diff
@@ -842,7 +827,7 @@ memblock_from_offset_opt(struct palloc_heap *heap, uint64_t off, int size)
 
 	m.type = off != 0 ? MEMORY_BLOCK_RUN : MEMORY_BLOCK_HUGE;
 #ifdef DEBUG
-	enum memory_block_type t = memblock_detect_type(&m, heap->layout);
```

Comment:
Can we fold this into the ASSERT? This way we can get rid of the ifdef.

Reply:
Sure.